pylint/test/functional/undefined_variable.py

```diff
@@ -175,3 +175,26 @@ def test2(self, is_disabled=disabled): # [undefined-variable]
 # See https://bitbucket.org/logilab/pylint/issue/111/
 try: raise IOError(1, "a")
 except IOError as err: print(err)
+
+
+def test_conditional_comprehension():
```

Comment:
What I thought was a solution added a regression where [this line](https://github.com/PyCQA/pylint/blob/dd15541d148d28d3a82639fed32f5f9f846e9832/pylint/checkers/design_analysis.py#L184) generated a pylint error.  This test verifies that no error is generated.